### DISCUSSION TITLE
canonspec description

### DISCUSSION THREAD

**Discussion by jtauber** [bible-technology/scripture-burrito#277] [Archive]
The canonSpec indicates for each relevant canon type (OT, DC, OTDC, NT) what tradition is followed:

e.g.

```
"canonSpec": {
  "ot": {
    "name": "western"
  },
  "nt": {
    "name": "western"
  }
},
```

This does not mean that every book in that tradition is included. That is covered by the `currentScope`.

Note that `ot` or `dc` may not be specified if `otdc` is present.

The `name` indicates a predefined tradition or names a new collection, in which case, the `books` array enumerates the collection.

e.g.

```
"canonSpec": {
  "ot": {
    "name": "western"
  },
  "nt": {
    "name": "x-matthewOnlyMillenialists",
    "books": ["MAT"]
  }
},
```

### Old Testament Canon Component

- `ethiopianProtestant`
- `syriac`
- `tanakh`
- `western`

### Old Testament + Deuterocanon Component

- `armenianApostolic`
- `armenianApostolic2`
- `armenianClassical`
- `catholicLxx`
- `catholicVulgate`
- `ethiopianOrthodox`
- `georgianOrthodox`
- `georgianOrthodox2`
- `greekOrthodox`
- `romanianOrthodox`
- `russianOrthodox`
- `russianProtestant`

### Deuterocanon Component 

- `armenianApostolic`
- `catholicAndAnglican`
- `catholicLxx`
- `catholicLxxSeparated`
- `catholicPlusLutheran`
- `catholicVulgate`
- `catholicVulgateSeparated`
- `czechKralicka`
- `danishLutheran`
- `ethiopianOrthodox`
- `georgianOrthodox`
- `georgianSynodal`
- `germanLutheran`
- `greekOrthodox`
- `kjv`
- `kjvNon`
- `paratext`
- `romanianOrthodox`
- `russianOrthodox`
- `russianSynodal`
- `turkishInterconfessional`
- `westernInterconfessional`
- `westernInterconfessional2`

### New Testament Canon Component

- `armenian`
- `ethiopianOrthodox`
- `ethiopianProtestant`
- `lutheran`
- `russian`
- `syriac`
- `western

### Old Testament Books

`GEN EXO LEV NUM DEU JOS JDG RUT 1SA 2SA 1KI 2KI 1CH 2CH EZR NEH EST JOB PSA PRO ECC SNG ISA JER LAM EZK DAN HOS JOL AMO OBA JON MIC NAM HAB ZEP HAG ZEC MAL`

### Deuterocanonical Books

`TOB JDT ESG WIS SIR BAR LJE S3Y SUS BEL 1MA 2MA 3MA 4MA 1ES 2ES MAN PS2 ODA PSS JSA JDB TBS SST DNT BLT EZA 5EZ 6EZ DAG PS3 2BA LBA JUB ENO 1MQ 2MQ 3MQ REP 4BA LAO`

### New Testament Books

`MAT MRK LUK JHN ACT ROM 1CO 2CO GAL EPH PHP COL 1TH 2TH 1TI 2TI TIT PHM HEB JAS 1PE 2PE 1JN 2JN 3JN JUD REV`

**Comment by jag3773** [comment]
Consensus is to remove this from the specification. If this is needed by any applications they can put it into their custom metadata.

**Reply by joelthe1** [reply]
Wanted to clarifying that we still need the USFM standard Book Codes to be used?

**Reply by jag3773** [reply]
Check to see if values of book codes are described as USFM book codes. These codes will be used at least in scope and role for ingredients. We should have a list of these in our schema for validation purposes.

**Comment by jtauber** [comment]
removed in #278

**Comment by jtauber** [comment]
`bookId` is defined in `common_schema.json` and enumerates the USFM book codes although does not call them that.

**Reply by jonathanrobie** [reply]
It would be good to call them that.

**Reply by jag3773** [reply]
Sounds good
